mm2src/mm2_main/tests/docker_tests/swap_watcher_tests.rs

```diff
@@ -0,0 +1,927 @@
+use crate::docker_tests::docker_tests_common::{eth_distributor, generate_eth_coin_with_random_privkey,
+                                               generate_jst_with_seed};
+use crate::integration_tests_common::*;
+use crate::{generate_utxo_coin_with_privkey, generate_utxo_coin_with_random_privkey, random_secp256k1_secret,
+            SecretKey};
+use coins::coin_errors::ValidatePaymentError;
+use coins::utxo::UtxoCommonOps;
+use coins::{FoundSwapTxSpend, MarketCoinOps, MmCoinEnum, SearchForSwapTxSpendInput, SendTakerPaymentArgs,
+            SendWatcherRefundsPaymentArgs, SwapOps, WatcherOps, WatcherValidateTakerFeeInput,
+            EARLY_CONFIRMATION_ERR_LOG, INVALID_RECEIVER_ERR_LOG, INVALID_SENDER_ERR_LOG, OLD_TRANSACTION_ERR_LOG};
+use common::{block_on, now_ms, DEX_FEE_ADDR_RAW_PUBKEY};
+use futures01::Future;
+use mm2_main::mm2::lp_swap::{dex_fee_amount_from_taker_coin, get_payment_locktime, MAKER_PAYMENT_SENT_LOG,
+                             MAKER_PAYMENT_SPEND_FOUND_LOG, MAKER_PAYMENT_SPEND_SENT_LOG,
+                             TAKER_PAYMENT_REFUND_SENT_LOG, WATCHER_MESSAGE_SENT_LOG};
+use mm2_number::BigDecimal;
+use mm2_number::MmNumber;
+use mm2_test_helpers::for_tests::{enable_eth_coin, eth_jst_conf, eth_testnet_conf, mm_dump, my_balance, mycoin1_conf,
+                                  mycoin_conf, start_swaps, MarketMakerIt, Mm2TestConf, WatcherConf,
+                                  DEFAULT_RPC_PASSWORD, ETH_SEPOLIA_NODE, ETH_SEPOLIA_SWAP_CONTRACT,
+                                  ETH_SEPOLIA_TOKEN_CONTRACT};
+use std::str::FromStr;
+use std::thread;
+use std::time::Duration;
+use uuid::Uuid;
+
+fn enable_eth_and_jst(mm_node: &MarketMakerIt) {
+    dbg!(block_on(enable_eth_coin(
+        mm_node,
+        "ETH",
+        ETH_SEPOLIA_NODE,
+        ETH_SEPOLIA_SWAP_CONTRACT,
+        Some(ETH_SEPOLIA_SWAP_CONTRACT)
+    )));
+
+    dbg!(block_on(enable_eth_coin(
+        mm_node,
+        "JST",
+        ETH_SEPOLIA_NODE,
+        ETH_SEPOLIA_SWAP_CONTRACT,
+        Some(ETH_SEPOLIA_SWAP_CONTRACT)
+    )));
+}
+
+fn get_balance(mm_node: &MarketMakerIt, ticker: &str) -> BigDecimal {
+    let json = block_on(my_balance(&mm_node, ticker));
+    let number = json["balance"].as_str().unwrap();
+    BigDecimal::from_str(number).unwrap().with_scale(2)
+}
```

Comment:
As `structs.rs` were moved to `mm2_test_helpers` crate recently, we can start refactoring RPC calling functions to return specific type. E.g. `pub async fn my_balance(mm: &MarketMakerIt, coin: &str) -> MyBalanceResponse`.